dace/codegen/targets/intel_fpga.py

```diff
@@ -1691,7 +1700,9 @@ def visit_Name(self, node):
              or defined_type == DefinedType.StreamArray) and memlet.dynamic):
             # Input memlet, we read from channel
             # we should not need mangle here, since we are in a tasklet
-            updated = ast.Name(id="read_channel_intel({})".format(node.id))
+            updated = ast.Call(func=ast.Name(id="read_channel_intel"),
```

Comment:
I had code that looked like this:

```python
if some_condition:
  some_tmp_var = read_channel_intel(connector_name)
  do_something_with_tmp_var(some_tmp_var)
```

This did not compile, because type inference does not understand that `read_channel_intel` returns the type of the channel object. With the update, type inference recognizes this case, and returns the type of the channel!